recipy/log.py

```diff
@@ -293,17 +297,44 @@ def output_file_diffs():
     if not option_set('data', 'file_diff_outputs'):
         return
 
-    db = open_or_create_db()
-    diffs_table = db.table('filediffs')
-    diffs = diffs_table.search(Query().run_id == RUN_ID)
+    encodings = ['utf-8', 'latin-1']
+
+    with open_or_create_db() as db:
+        diffs_table = db.table('filediffs')
+        diffs = diffs_table.search(Query().run_id == RUN_ID)
+
     for item in diffs:
-        diff = difflib.unified_diff(open(item['tempfilename']).readlines(),
-                                    open(item['filename']).readlines(),
-                                    fromfile='before this run',
-                                    tofile='after this run')
-        diffs_table.update({'diff': ''.join([l for l in diff])},
-                           eids=[item.eid])
+        if option_set('general', 'debug'):
+            print('Storing file diff for "%s"' % item['filename'])
+
+        lines1 = None
+        lines2 = None
+        for enc in encodings:
+            try:
+                with codecs.open(item['tempfilename'], encoding=enc) as f:
+                    lines1 = f.readlines()
+            except UnicodeDecodeError:
+                pass
+
+            try:
+                with codecs.open(item['filename'], encoding=enc) as f:
+                    lines2 = f.readlines()
+            except UnicodeDecodeError:
+                pass
+
+        if lines1 is not None and lines2 is not None:
```

Comment:
Actually, forget that point. I see what you are doing here, you might have empty files.